service/history/replication/task_hydrator.go

```diff
@@ -275,3 +283,47 @@ func (l mutableStateLoader) GetMutableState(ctx context.Context, domainID, workf
 func timeToUnixNano(t time.Time) *int64 {
 	return common.Int64Ptr(t.UnixNano())
 }
+
+type immediateHistoryProvider struct {
+	blob     *persistence.DataBlob
+	nextBlob *persistence.DataBlob
+}
+
+func (h immediateHistoryProvider) GetEventBlob(ctx context.Context, task persistence.ReplicationTaskInfo) (*types.DataBlob, error) {
```

Comment:
nit/optional: explicitly tell that args are not used
```suggestion
func (h immediateHistoryProvider) GetEventBlob(_ context.Context, _ persistence.ReplicationTaskInfo) (*types.DataBlob, error) {
```